bpf/lib/eps.h

```diff
@@ -45,13 +45,14 @@ lookup_ip4_endpoint(struct iphdr *ip4)
 	return map_lookup_elem(&cilium_lxc, &key);
 }
 
-#if (defined POLICY_EGRESS || defined POLICY_INGRESS) && defined LXC_ID
 /* IPCACHE_STATIC_PREFIX gets sizeof non-IP, non-prefix part of ipcache_key */
 #define IPCACHE_STATIC_PREFIX							\
 	(8 * (sizeof(struct ipcache_key) - sizeof(struct bpf_lpm_trie_key)	\
 	      - sizeof(union v6addr)))
 #define IPCACHE_PREFIX_LEN(PREFIX) (IPCACHE_STATIC_PREFIX + PREFIX)
 
+#define V6_CACHE_KEY_LEN IPCACHE_PREFIX_LEN(sizeof(union v6addr)*8)
```

Comment:
I suspect that the existing declarations of these were wrong, as `ipcache_lookup[46]()` already apply `IPCACHE_PREFIX_LEN(...)` to the specified prefix.

I think that this, and the existing declarations shouldn't be `IPCACHE_PREFIX_LEN(...)`, just `(sizeof(union v6addr)*8)`. (The 8 indicates translating the sizeof as bytes into bits)

Same for V4 equivalent.

Reply:
Does this mean ipcache was broken on more recent kernels? It must be.

Reply:
Given that we don't regularly test on newer kernels, quite possible.